ComponentKit/Core/Scope/CKComponentScopeFrame.mm

```diff
@@ -82,6 +83,75 @@ + (CKComponentScopeFramePair)childPairForPair:(const CKComponentScopeFramePair &
+     not added to the component hierarchy) this behavior guarantees that component 4 always acquires the same component
+     scope.
+
+     Compare this behavior to the behavior that would exist if the initial component scope tree were not shared:
+
+                                       +-------+         +-------+         +-------+
+                                       |       |         |       |         |       |
+                                       |   A   |         |   A   |         |   A   |
+                                       |      1|         |      2|         |      3|
+                                       +-------+         +-------+         +-------+
+                                           |                 | collision       | collision
+                                           +-----------------+-----------------+
+                                          /|\                .                /|\
+                                         / | \               .               / | \
+                                        /  |  \              .              /  |  \
+                                   +---+ +---+ +---+                   +---+ +---+ +---+
+                                   | 1 | | 2 | | 3 |                   | 1'| | 2'| | 3'|
+                                   +---+ +---+ +---+                   +---+ +---+ +---+
+                                    / \                                 / \
+                                +---+ +---+                         +---+ +---+
+                                | 4 | | 5 |                         | 4'| | 5'|
+                                +---+ +---+                         +---+ +---+
+
+     Each component scope frame participating in the collision now has its own unique component scope tree. For
+     component scope frame A1 the outcome is largely the same. Things get a bit more interesting for A2 and A3. Notice
+     that A3 now has its own, nearly identical, component scope tree. The structure is the same but the component state
+     and component controllers are different.
+
+     Problems arise when the component that owns component scope frame A3 is added to the component hierarchy. Suppose
+     A3 is building its component scope tree for the first time. The component that owns component scope frame 4' will
+     acquire a new component controller as there is no equivalent previous frame, as expected.
+
+     The next time the component hierarchy is created (e.g. after a component state update) component scope frame 4'
+     actually finds component scope frame 4 in the equivalent previous frame. This means component 4' will acquire a
+     DIFFERENT component controller instance than it had originally. Why? Because the component scope frame above
+     component scope frame A will only ever have A1 as a child because A1 was inserted before A2 and A3.
+     */
+    newChild->_children = existingChild->second->_children;
```

Comment:
This doesn't feel terribly efficient but the circumstances where the children need to be copied should be relatively rare.